zio-http-benchmarks/src/main/scala/zhttp.benchmarks/CookieDecodeBenchmark.scala

```diff
@@ -0,0 +1,31 @@
+package zhttp.benchmarks
+
+import org.openjdk.jmh.annotations._
+import zhttp.http._
+
+import java.time.Instant
+import java.util.concurrent.TimeUnit
+
+@State(Scope.Thread)
+@BenchmarkMode(Array(Mode.Throughput))
+@OutputTimeUnit(TimeUnit.SECONDS)
+class CookieDecodeBenchmark {
+  private val cookie    = Cookie(
+    "CookiesName",
+    "CookiesValue",
+    Some(Instant.now()),
+    Some("domainValue"),
+    Some(Path("Some", "Path")),
+    true,
+    true,
+    Some(1200),
+    Some(Cookie.SameSite.Strict),
+  )
```

Comment:
Can we use a strings that are at least 100 chars or more. 

```scala
val random = new scala.util.Random((Math.random * Int.MaxValue).toInt)
val name = random.alphanumeric.take(100).mkString("")
val value = random.alphanumeric.take(100).mkString("")
val domain = random.alphanumeric.take(100).mkString("")
val name = Path((0 to 10).map { _ => random.alphanumeric.take(10).mkString("") })
```

The real improvement in performance will be observed in larger strings.

Reply:
Updated benchmark code with the following results:
```
Old code:
(min, avg, max) = (393771,450, 395268,674, 396456,159), stdev = 1368,893

New code:
(min, avg, max) = (634553,745, 636047,487, 637677,233), stdev = 1566,179

```